lib/monitor.js

```diff
@@ -121,31 +121,37 @@ internals.Monitor.prototype.start = function (callback) {
         };
     };
 
+    var reportersIndex = 0;
+
     Items.serial(this.settings.reporters, function (item, next) {
 
-        var reporter;
+        var hint = self.settings.reporters.length > 1 ? '(' + reportersIndex++ + ')' : '';
 
-        // If it has a reporter constructor, then create a new one, otherwise, assume it is
-        // a valid pre-constructed reporter
-        if (item.reporter) {
+        // If the supply a path or module node, try to load it
+        if (typeof item.reporter === 'string') {
+            item.reporter = require(item.reporter);
+        }
 
-            // If the supply a path or module node, try to load it
-            if (typeof item.reporter === 'string') {
-                item.reporter = require(item.reporter);
-            }
+        Hoek.assert(typeof item.reporter === 'function', 'reporter must be a constructor function ', hint);
 
-            Hoek.assert(typeof item.reporter === 'function', 'reporter key must be a constructor function');
-            Hoek.assert(typeof item.events === 'object', 'reporter must specify events to filter on');
+        var attributes = item.reporter.attributes;
+        Hoek.assert(attributes && typeof attributes === 'object', 'reporter constructor missing attributes object', hint);
 
-            var Reporter = item.reporter;
+        var metadata = {
+            name: attributes.name || Hoek.reach(attributes, 'pkg.name'),
+            version: attributes.version || Hoek.reach(attributes, 'pkg.version') || '0.0.0'
+        };
 
-            reporter = new Reporter(item.events, item.config);
-        }
-        else {
-            reporter = item;
-        }
+        Hoek.assert(metadata.name, 'reporter attributus missing name', hint);
+        Hoek.assert(typeof item.events === 'object', 'reporter must specify events to filter on');
 
+        var Reporter = item.reporter;
+        var reporter = new Reporter(item.events, item.config);
         Hoek.assert(reporter.init, 'Every reporter object must have an init method');
+
+        item.instance = reporter;
```

Comment:
I used the stashed instance for testing but I will rework the tests to not require this. I'll go ahead and drop these references.